core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -621,10 +621,10 @@ class KafkaZkClient private[zk] (zooKeeperClient: ZooKeeperClient, isSecure: Boo
         case Code.NONODE => None
         case _ => throw getDataResponse.resultException.get
       }
-    }.map(_.get)
-      .map(topicIdAssignment => (topicIdAssignment.topic,
-        topicIdAssignment.topicId.getOrElse(
-          throw new IllegalStateException("Topic " + topicIdAssignment.topic + " does not have a topic ID."))))
+    }.filter(_.isDefined)
```

Comment:
nit: getDataResponses.map(xxx).filter(_.isDefined).map(_.get) can be replaced by getDataResponses.flatMap(xxx)

Reply:
This will change soon with the new PR I'll be opening tomorrow that gates topic IDs behind IBP 2.8.

Reply:
Decided to just open it now. https://github.com/apache/kafka/pull/9814/files